src/main/java/saigonwithlove/ivy/intellij/settings/IvyModuleListener.java

```diff
@@ -0,0 +1,37 @@
+package saigonwithlove.ivy.intellij.settings;
+
+import static saigonwithlove.ivy.intellij.settings.PreferenceService.State;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.project.ModuleListener;
+import com.intellij.openapi.project.Project;
+import org.jetbrains.annotations.NotNull;
+import saigonwithlove.ivy.intellij.shared.IvyModule;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class IvyModuleListener implements ModuleListener {
+  @Override
+  public void moduleRemoved(@NotNull Project project, @NotNull Module module) {
+    PreferenceService preferenceService = project.getService(PreferenceService.class);
+    List<IvyModule> initialModulesCopy =
+        new ArrayList<>(preferenceService.getState().getIvyModules());
+    List<IvyModule> remainingModules =
+        initialModulesCopy.stream()
+            .filter(item -> !item.getName().equalsIgnoreCase(module.getName()))
+            .collect(Collectors.toList());
+    preferenceService.update(
+        (oldState -> {
+          State newState = new State();
```

Comment:
Hi @zadewu, no need to create new state. you only need to `setIvyModules`. I checked the implementation, it will not differentiate the state.

For example:

```
        preferenceService.update(
            state -> {
              configuration.setValue(newValue);
              state.getServerProperties().put(configuration.getName(), configuration);
              return state;
            });
```

Reply:
Hi @cavoirom, I adjust the code and also do the test. It can run normally